rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/widget/CovidEditTextFactoryTest.java

```diff
@@ -0,0 +1,88 @@
+
+import com.rengwuxian.materialedittext.MaterialEditText;
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.fragments.JsonFormFragment;
+import com.vijay.jsonwizard.interfaces.CommonListener;
+import com.vijay.jsonwizard.utils.FormUtils;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.robolectric.util.ReflectionHelpers;
+
+import java.util.List;
+
+import io.ona.rdt.R;
+import io.ona.rdt.robolectric.widget.WidgetFactoryRobolectricTest;
+
+public class CovidEditTextFactoryTest extends WidgetFactoryRobolectricTest {
+
+    private static final String RIGHT_MARGIN = "right_margin";
+    private static final String EXPECTED_MARGIN = "5dp";
+    private static final String EXPECTED_PADDING = "10dp";
+
+    CovidEditTextFactory covidEditTextFactory;
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        covidEditTextFactory = new CovidEditTextFactory();
+    }
+
+    @Test
+    public void testEditTextShouldReturnUpdatedProperties() throws Exception {
+        RelativeLayout rootLayout = getRootLayout();
+        MaterialEditText editText = getEditText(rootLayout);
+        Context context = editText.getContext();
+
+        // verify root layout minimum height
+        Assert.assertEquals(0, rootLayout.getMinimumHeight());
+
+        // verify floating label type
+        Assert.assertFalse(ReflectionHelpers.getField(editText, "floatingLabelEnabled"));
+        Assert.assertFalse(ReflectionHelpers.getField(editText, "highlightFloatingLabel"));
+
+        // verify margins
+        RelativeLayout.LayoutParams layoutParams = (RelativeLayout.LayoutParams) editText.getLayoutParams();
+        Assert.assertEquals(FormUtils.getValueFromSpOrDpOrPx(EXPECTED_MARGIN, context), layoutParams.leftMargin);
+        Assert.assertEquals(FormUtils.getValueFromSpOrDpOrPx(EXPECTED_MARGIN, context), layoutParams.topMargin);
+        Assert.assertEquals(FormUtils.getValueFromSpOrDpOrPx(EXPECTED_MARGIN, context), layoutParams.rightMargin);
+        Assert.assertEquals(FormUtils.getValueFromSpOrDpOrPx(EXPECTED_MARGIN, context), layoutParams.bottomMargin);
+
+        // verify floating label padding
+        Assert.assertEquals(FormUtils.getValueFromSpOrDpOrPx(EXPECTED_PADDING, context), editText.getFloatingLabelPadding());
+    }
+
+    private RelativeLayout getRootLayout() throws Exception {
+        List<View> views = covidEditTextFactory.getViewsFromJson(JsonFormConstants.STEP1, jsonFormActivity,
+                Mockito.mock(JsonFormFragment.class), getWidget(), Mockito.mock(CommonListener.class));
+        return (RelativeLayout) views.get(0);
+    }
+
+    private MaterialEditText getEditText(RelativeLayout rootLayout) throws Exception {
+        return (MaterialEditText) ((RelativeLayout) rootLayout.findViewById(R.id.edit_text_layout)).getChildAt(0);
```

Comment:
Iterate over the child elements and use the type of  child element to extract the edit text instead, in case the position of the edit text changes.

Reply:
fixed.